### PR TITLE
Add HTML Language

### PR DESCRIPTION
**Pull request by MrIntellectual**

### CODE REVIEW

**Comment by whippingdot** [comment]
This done or a draft / WIP?

File: README.md
Comment (whippingdot) [inline README.md:11]: Hey add a ` \` at the end of this Cookeylang line. This is to create a new line.

File: HTML/1. intro.md
Comment (whippingdot) [inline HTML/1. intro.md:13]: This is wrong. There are some elements which don't have an ending tag and there are some elements without an ending tag which have a '/>' in their starting tag / only tag itself.

Comment (MrIntellectual) [reply HTML/1. intro.md:13]: K forgot about that I'll add that, also It's me IntellectualGuy

Comment (whippingdot) [reply HTML/1. intro.md:13]: O hi. I didn't even realize, I thought you were some random dude lul. Thanks for adding a PR on this.

One note if you look at my C++ files you will see a pattern I use. Yours is pretty similar to that, but it would be nicer if you could use that same pattern. If you don't want to, its ok and I won't mind.

Comment (MrIntellectual) [reply HTML/1. intro.md:13]: Ok I'll check that out, also I don't mind if anyone else helps. And forgive if I ask anything stupid, I just joined github yesterday so I'm new to this.

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:5]: The use of "normal programming language" here suggests that HTML can be compared to programming languages, as if HTML were a programming language, just not a "normal" one, contradicting what was said before, that it is a markup language.

I'd suggest just removing the "normal."

Also, it may be better to refer to Python as a "scripting language," but whatever.

File: README.md
Comment (whippingdot) [inline README.md:12]: bruh not this line just the Cookeylang line

File: HTML/1. intro.md
Comment (CrimsonCodes0) [reply HTML/1. intro.md:13]: Do note that writing elements like "input" as `<input />` is incorrect, as that is XML syntax, not HTML, the correct way would be `<input>`, merely without an ending tag.

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:22]: You never introduced the "doctype" element, and many web developers have no clue what it does.

File: HTML/2. headings.md
Comment (CrimsonCodes0) [inline HTML/2. headings.md:5]: These shouldn't be categorized by size, but by semantics and meaning, considering that CSS can completely invert the sizes, make them all equal, etc.

Comment (CrimsonCodes0) [reply HTML/2. headings.md:5]: Oh, this is mentioned afterwards, how about my mixing them, or deferring focus on the sizes?

File: HTML/1. intro.md
Comment (CrimsonCodes0) [reply HTML/1. intro.md:22]: The definition that was recently added is incorrect, you may want to research what it does yourself, look into on Wikipedia and MDN.

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:15]: Maybe add examples of some void elements?

And remove the second structure, it's not HTML, it just works, since HTML doesn't throw exceptions or anything of the sort.

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:5]: Markup languages just markup text, they aren't necessarily for the web at all.

```suggestion
HTML is a markup language that is used to create webpages, so it's not like a programming language like Python
```

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:9]: How about, instead of introducing `<element>...</element>` as the normal syntax, and void elements (`<element>`) as an exception, just introducing them side-by-side, with their appropriate syntaxes, more as equals?

File: HTML/3. lists.md
Comment (CrimsonCodes0) [inline HTML/3. lists.md:1]: If you want, you could toss in `<dl>`.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

File: HTML/4. text.md
Comment (CrimsonCodes0) [inline HTML/4. text.md:24]:
```suggestion
You can do Italics through two tags, the `<i>` tag or the `<em>` tag, and `<em>` is semantically correct.
```

File: HTML/4. text.md
Comment (CrimsonCodes0) [inline HTML/4. text.md:70]:
```suggestion
Superscript and Subscript are done through the `<sub>` and `<sup>` tags.
```

File: HTML/4. text.md
Comment (CrimsonCodes0) [inline HTML/4. text.md:86]: "goes" doesn't make sense here, maybe you know what fits best?

File: HTML/4. text.md
Comment (CrimsonCodes0) [inline HTML/4. text.md:91]:
```suggestion
The ending `</strong>` tag should come before the ending `</p>` tag.
```
I feel like this is more intuitive, and helps reader catch which tag you are referring to much faster.

**Review by CrimsonCodes0 — COMMENTED** [review]
I left a few suggestions that should help improve the pages, if you disagree, feel free to point out why.

File: HTML/3. lists.md
Comment (whippingdot) [reply HTML/3. lists.md:1]: Ya wait I learned about `<dl>`s in school so I think it should be mentioned...

P.S. How do you suggest a change. When I review I can suggest a change but that doesn't actually let you add the change directly. It just adds a bunch of comments.

Comment (CrimsonCodes0) [reply HTML/3. lists.md:1]: Well, you seem to know how to suggest a change already?

CTRL+G, or clicking the icon with an underlined plus-minus.

Comment (whippingdot) [reply HTML/3. lists.md:1]:
```suggestion
DAFSLKJSDFLKASD
```

Comment (MrIntellectual) [reply HTML/3. lists.md:1]: Yeah okay I'm planning on a webpage with tables, and description lists

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:7]: They aren't different types of elements, they're all elements. It's just the syntax that is different.

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:12]: "Self's closing" -> "self-closing"

**Comment by darkdarcool** [comment]
whoops, didn't mean to do that, srry. Just ignore that.

**Comment by CrimsonCodes0** [comment]
@darkdarcool Why don't you undo the action then?

**Comment by darkdarcool** [comment]
thx @whippingdot. It wouldn't give me a button to undo it.(or im dumb and couldn't find it)

**Comment by whippingdot** [comment]
When not a draft PR, it is under the reviewers section at the top.

**Comment by darkdarcool** [comment]
oh ok

**Comment by darkdarcool** [comment]
Can we add ID attributes in the md's? We can loosely cover the js for that, like a very short and simple explanation of 
``` js
document.getElementByID("id_name").innerHTML = "eeee new text eeee"
```

**Comment by MrIntellectual** [comment]
@darkdarcool 

Maybe a DOM/JQuery tutorial would be better suited for that

**Comment by MrIntellectual** [comment]
@whippingdot It's ready for review, do you mind checking it for any issues and merge it?

**Comment by darkdarcool** [comment]
> @darkdarcool
> 
> Maybe a DOM/JQuery tutorial would be better suited for that

Hmmm, maybe(probably). But maybe you can skim over that very lightly? It just seems like something that you can spend like a small time on the readme on. Then you can dedicate multiple MD's to DOM/JQuery or something.

**Comment by CrimsonCodes0** [comment]
@darkdarcool Really, I feel like JavaScript shouldn't be mentioned at this point, it wouldn't fit into the tutorial yet, especially considering how CSS hasn't been brought into it yet, and should come far before JS. (And maybe how there isn't a JS tutorial yet?)

Also, the `document` and it's methods access the abstract DOM, not HTML, so it is way too early.

Also, JQuery has really lost traction in the last few decades, and isn't used that often.

**Comment by CrimsonCodes0** [comment]
I'd suggest keeping the languages separate, there could be a separate tutorial for HTML, CSS, SASS, SCSS, JS, etc.

Quite a lot can be done with plain HTML, think interactivity, forms, buttons, menus, etc.

The others would rely on HTML, but not vice-versa.

@whippingdot What do you think?

**Comment by darkdarcool** [comment]
@CrimsonCodes0, sounds like a good idea to me!

**Comment by whippingdot** [comment]
Hey @MrIntellectual I didn't see that this was ready for review - will review it today if I have the time.

File: HTML/1. intro.md
Comment (CrimsonCodes0) [inline HTML/1. intro.md:9]:
```suggestion
1. The first way of declaring elements is with an opening tag and a closing tag, the opening tag has a `<` to begin and a `>` to end. The closing tag has a `</` to begin and a `>` to end. Everything else between the opening and closing tags is either content or other nested elements
```

`<html>...</html>`

**Review by CrimsonCodes0 — CHANGES_REQUESTED** [review]
Small error in syntax.

File: HTML/1. intro.md
Comment (whippingdot) [reply HTML/1. intro.md]: Eyyy I am succesfull at glitching github

I committed the suggestion but it actually isn't committed sry lul

You still have to fix this @MrIntellectualGuy

Comment (whippingdot) [reply HTML/1. intro.md]: Actually, I did commit it

how
wut
yay
les
go

@CrimsonCodes0 I committed your suggestion somehow

**Comment by whippingdot** [comment]
I fixed ur feedback @CrimsonCodes0

**Comment by whippingdot** [comment]
Not reviewed yet - I will like tomorrow or today (i have went somewhere else, sorry)

File: HTML/1. intro.md
Comment (whippingdot) [inline HTML/1. intro.md:6]: Try to not have these line breaks **next time**

File: HTML/1. intro.md
Comment (whippingdot) [inline HTML/1. intro.md:5]: Remove the like Python

File: HTML/1. intro.md
Comment (whippingdot) [inline HTML/1. intro.md:12]: Can also end with /> I am pretty sure... or it starts with </

File: HTML/1. intro.md
Comment (whippingdot) [inline HTML/1. intro.md:13]: Edit these line breaks and change them into subheadings for easier navigation

File: HTML/1. intro.md
Comment (whippingdot) [inline HTML/1. intro.md:23]: Bruh...`<body>`

File: HTML/3. lists.md
Comment (whippingdot) [inline HTML/3. lists.md:6]: ?? Why the lines?

File: HTML/3. lists.md
Comment (whippingdot) [inline HTML/3. lists.md:34]: Again, I would tell you to use sub/sub-sub headings before the code blocks and the outputs, but it is your wish

File: HTML/4. text.md
Comment (whippingdot) [inline HTML/4. text.md:24]: Again talk about how em is used to emphasize something while i is just used to make something italic

File: HTML/4. text.md
Comment (whippingdot) [inline HTML/4. text.md:39]: I don't know what ins means, but if it has a meaning state that too

File: HTML/4. text.md
Comment (whippingdot) [inline HTML/4. text.md:54]: Again I don't know what del is but I guess it is delete and I don't know what s is. If they have meanings the state them

File: HTML/4. text.md
Comment (whippingdot) [inline HTML/4. text.md:69]: `##`

File: HTML/4. text.md
Comment (whippingdot) [inline HTML/4. text.md:84]: `##`

File: HTML/5. attributes.md
Comment (whippingdot) [inline HTML/5. attributes.md:5]:
```suggestion
1.	The first type of attributes contains an attribute value. For example, there are the `length` and `width` attributes. They specify the length, and width of an element.
```

File: HTML/5. attributes.md
Comment (whippingdot) [inline HTML/5. attributes.md:38]:
```suggestion
Output:
```
Why the name change?

**Review by whippingdot — CHANGES_REQUESTED** [review]
I left a few suggestions. For some time, I couldn't suggest changes as I forgot how to, but afterwards I remembered and you can see like two times where you can just press the commit button to make changes.

**Comment by whippingdot** [comment]
Heyyyyyyyy @MrIntellectual make a few changes, I will review those, and then I can fix the merge conflicts and MERGE!! 🥳🥳

**Comment by darkdarcool** [comment]
@whippingdot Wahoo!

File: HTML/1. intro.md
Comment (CrimsonCodes0) [reply HTML/1. intro.md:12]: An element can end with `/` just as much as `/` can be an attribute name: it's only allowed because HTML normalizes all errors.

`<a /="b">` - clearly incorrect.

The `/` in `/>` is in a boolean attribute position.

**Comment by MrIntellectual** [comment]
@whippingdot Fixed Them

**Review by darkdarcool — APPROVED** [review]
All this is great!

**Review by whippingdot — CHANGES_REQUESTED** [review]
I think you just have to make these two changes and that is all.

I am not looking at the output of the files yet so I might make some of my own changes after merging, but I will keep it to like 1 or 2 cause I want this to be in your style, not mine

File: HTML/2. headings.md
Comment (MrIntellectual) [reply HTML/2. headings.md:44]: Yeah, that's what I was thinking, but when I tried to put them on separate lines it automatically made a small line break, and I thought that would look weird.

Comment (whippingdot) [reply HTML/2. headings.md]: Fine, then ima merge now!

**Review by whippingdot — APPROVED** [review]

**Pull request merged**
